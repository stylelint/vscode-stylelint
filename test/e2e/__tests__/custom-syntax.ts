import path from 'path';
import pWaitFor from 'p-wait-for';
import { commands } from 'vscode';

describe('"stylelint.customSyntax" setting', () => {
	it('should auto-fix using the specified custom syntax', async () => {
		const { document } = await openDocument(path.resolve(workspaceDir, 'custom-syntax/test.css'));

		await pWaitFor(
			async () => {
				const names = await commands.getCommands();

				return (
					// cspell:disable-next-line
					names.includes('stylelint.executeAutofix') && names.includes('stylelint.applyAutoFix')
				);
			},
			{ timeout: 2000 },
		);

		// cspell:disable-next-line
		await commands.executeCommand('stylelint.executeAutofix');

		expect(document.getText()).toMatchSnapshot();
	});
});
