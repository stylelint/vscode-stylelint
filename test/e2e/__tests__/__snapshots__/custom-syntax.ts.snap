// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"stylelint.customSyntax" setting should auto-fix using the specified custom syntax 1`] = `
"/* prettier-ignore */
.foo .bar
  color: rgb(0 0 0)"
`;
